Sources/Networking/HTTPClient/HTTPClient.swift

```diff
@@ -531,3 +527,18 @@ extension HTTPResponse {
     }
 
 }
+
+private extension HTTPResponse where Body == Data {
+
+    func parseUnsuccessfulResponse() -> NetworkError {
+        let isJSON = self.value(forHeaderField: HTTPClient.ResponseHeader.contentType.rawValue) == "application/json"
```

Comment:
My guess is yes but should we confirm the response will always have this header? (When the server is up)

Reply:
If it doesn't we still produce an error, we just won't try to parse the specific backend error, so in practice it doesn't really matter that much.